### PR TITLE
changes to u_milestone.py

### PR DESCRIPTION
**Pull request by dy2403**
Hi @sbillinge !
Now, when the user informs only "20dy" (for example), it returns all the projectum ids that contain "20dy"! 
Also, when verbose is not informed, it returns a simple list, with the name of the milestone, due date and status!
Can you please review it?

### CODE REVIEW

File: regolith/helpers/u_milestonehelper.py [hunk 1]
```diff
@@ -13,28 +13,29 @@
 ALLOWED_STATI = {"p":"proposed", "s":"started", "f":"finished", "b":"back_burner","c":"converged"}
 
 def subparser(subpi):
-    subpi.add_argument("projectum_id", help="the id of the projectum")
+    subpi.add_argument("projectum_id", help="The id of the projectum.")
     subpi.add_argument("-v", "--verbose", action="store_true",
-                        help="gives a list of the milestones available to update.")
+                        help="Gives a list of the milestones available to update "
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:18]: "Increases the verbosity of the output."

File: regolith/helpers/u_milestonehelper.py [hunk 2]
```diff
@@ -13,28 +13,29 @@
...
                         type = int)
     subpi.add_argument("-d", "--due_date",
-                       help="new due date of the milestone in ISO format (YYYY-MM-DD) "
-                            "required to add a new milestone")
+                       help="New due date of the milestone in ISO format(YYYY-MM-DD). "
+                            "Required to add a new milestone. ")
     subpi.add_argument("-n", "--name",
-                       help="name of the new milestone. "
-                            "required to add a new milestone.")
+                       help="Name of the new milestone. "
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:27]: "Name of the milestone. Required for a new milestone."

File: regolith/helpers/u_milestonehelper.py [hunk 3]
```diff
@@ -13,28 +13,29 @@
...
+                             "with specific information about each milestone.")
     subpi.add_argument("-i", "--index",
-                        help="index of the item in the enumerated list to update",
+                        help="Index of the item in the enumerated list to update. ",
                         type = int)
     subpi.add_argument("-d", "--due_date",
-                       help="new due date of the milestone in ISO format (YYYY-MM-DD) "
-                            "required to add a new milestone")
+                       help="New due date of the milestone in ISO format(YYYY-MM-DD). "
+                            "Required to add a new milestone. ")
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:25]: "Required for a new milestone."
Please use this wording consistently.

File: regolith/helpers/u_milestonehelper.py [hunk 4]
```diff
@@ -13,28 +13,29 @@
...
     subpi.add_argument("-n", "--name",
-                       help="name of the new milestone. "
-                            "required to add a new milestone.")
+                       help="Name of the new milestone. "
+                            "Required to add a new milestone.")
     subpi.add_argument("-o", "--objective",
-                       help="objective of the new milestone. "
-                            "required to add a new milestone")
+                       help="Objective of the new milestone. "
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:30]: "Objective of the milestone"

File: regolith/helpers/u_milestonehelper.py [hunk 5]
```diff
@@ -13,28 +13,29 @@
...
+                            "Required to add a new milestone.")
     subpi.add_argument("-s", "--status",
-                       help="status of the milestone/deliverable: "
-                            f"{ALLOWED_STATI}")
+                       help="Status of the milestone/deliverable: "
+                            f"{ALLOWED_STATI}.")
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:34]: please add "Defaults to ??? for a new milestone"     Please also enter this information below for other args as needed.

File: regolith/helpers/u_milestonehelper.py [hunk 6]
```diff
@@ -13,28 +13,29 @@
...
     subpi.add_argument("-t", "--type",
-                       help="type of the milestone: "
+                       help="Type of the milestone: "
                             f"{ALLOWED_TYPES}"
-                            "new milestone defaults to meeting")
+                            "New milestone defaults to meeting.")
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:38]: see above

File: regolith/helpers/u_milestonehelper.py [hunk 7]
```diff
@@ -70,7 +71,14 @@ def db_updater(self):
         filterid = {'_id': key}
         target_prum = rc.client.find_one(rc.database, rc.coll, filterid)
         if not target_prum:
-            raise RuntimeError(f"Cannot find {key} in the projecta collection")
+            pra = fragment_retrieval(self.gtx["projecta"], ["_id"], rc.projectum_id)
+            if len(pra) == 0:
+                raise RuntimeError("Please input a valid projectum id or a valid fragment of a projectum id")
+            print("Projectum not found. Projectum with similar names: ")
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:77]: Projectum -> Projecta

File: regolith/helpers/u_milestonehelper.py [hunk 8]
```diff
@@ -70,7 +71,14 @@ def db_updater(self):
...
+            for i in range(len(pra)):
+                print(f"{pra[i].get('_id')}")
+            print("Please rerun the helper with the correct name of the projectum.")
```
Comment (sbillinge) [inline regolith/helpers/u_milestonehelper.py:80]: "Please rerun the helper specifying the complete ID."

Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 7 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 6 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 5 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 4 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 3 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 2 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

File: regolith/helpers/u_milestonehelper.py (hunk 1 above)
Comment (dy2403) [reply regolith/helpers/u_milestonehelper.py]: done!

**Comment by dy2403** [comment]
@sbillinge ready to review and merge!

**Comment by dy2403** [comment]
@sbillinge I have a question about news!
In this case, I wrote a news when this u_milestone was first merged.
Now, I fixed some bugs. 
Should I create a new file with the corrections of the bugs, or should I update the file I already have (in the fixed field)?

**Comment by sbillinge** [comment]
> 
> 
> @sbillinge I have a question about news!
> In this case, I wrote a news when this u_milestone was first merged.
> Now, I fixed some bugs.
> Should I create a new file with the corrections of the bugs, or should I update the file I already have (in the fixed field)?

in this case we probably don't need a news.  If you are fixing bugs for a feature that hasn't been released it doesn't make that much sense to say "fixing bug".  Maybe read the original news and update if it is not correctly describing new functionality so that it describes the function properly.  Let me know when it is done and I can merge

**Comment by dy2403** [comment]
@sbillinge 
Ok! I think the current news correctly describe this helper:
a milestones helper for adding/updating milestones to the projecta collection.
Is it correct?

**Comment by sbillinge** [comment]
exciting, thanks @dy2403

**Pull request merged**